### PR TITLE
Switch to Github Actions for building and uploading bottles

### PR DESCRIPTION
**Pull request by dawidd6**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [ ] Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.

-----

How do you @Homebrew/linux folks feel about a change like that?

### CODE REVIEW

**Comment by dawidd6** [comment]
I would also like to report the status of build and upload steps through PR labels instead of creating status like @iMichka tried to do. I think it would be nicer to have it in workflow like that:
- build succeeded => "build/success"
- build succeeded with issues => "build/success-with-issues"
- build failure => "build/failure"
- upload succeeded => "upload/success"
- upload failure => "upload/failure"

These labels are just a proposal how they can be named.

**Comment by sjackman** [comment]
Cool! Nice work, Dawid! I'll defer to Michka and Issy on the details, but I like the general idea of moving from Azure to GitHub Actions.

File: .github/workflows/build_and_upload_bottles.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+              --keep-old
+          echo "::set-output name=build_result::$?"
+          ls *.bottle.*
+          echo "::set-output name=bottle_result::$?"
+          mv /bottles $GITHUB_WORKSPACE
+      - name: Upload bottles
+        id: upload
+        env:
+          HOMEBREW_BINTRAY_USER: linuxbrew
+          HOMEBREW_BINTRAY_KEY: ${{secrets.HOMEBREW_BINTRAY_KEY}}
```
Comment (MikeMcQuaid) [inline .github/workflows/build_and_upload_bottles.yml]: This secret will not be exposed to forks so cannot be used to upload from here, unfortunately.

**Comment by MikeMcQuaid** [comment]
> I would also like to report the status of build and upload steps through PR labels instead of creating status like @iMichka tried to do. I think it would be nicer to have it in workflow like that:

I think CI status is better than labels, personally. What would labels solve that CI status does not?

File: .github/workflows/build_and_upload_bottles.yml (hunk 1 above)
Comment (dawidd6) [reply .github/workflows/build_and_upload_bottles.yml]: You're right, just tested it:
  branch: https://github.com/dawidd6/homebrew-test/commit/4c5cc54eaa12e4c6bf0dfd7582aa12b8a45ae168/checks?check_suite_id=281008553
  fork: https://github.com/dawidd6/homebrew-test/commit/b548a29ce893343d8051687edf4ef9f7ad7cf8ef/checks?check_suite_id=281013405

Also, relevant issue on GH Forum: https://github.community/t5/GitHub-Actions/Make-secrets-available-to-builds-of-forks/m-p/30678

Any ideas how to resolve this?

Comment (MikeMcQuaid) [reply .github/workflows/build_and_upload_bottles.yml]: There is no solution, unfortunately. You'd need to wait for GitHub Actions to add this. Note that even if they do that's probably not good enough as e.g. they could use this to send `BINTRAY_KEY` to their own server. What you need (which Azure Pipelines's release pipeline and Jenkins multiple-jobs provide) is a way to make the bottle upload happen in a workflow that users cannot modify as part of their fork. Maybe there's some way to trigger a job on another repo or something?

**Comment by MikeMcQuaid** [comment]
> This allows me (and others) to easily execute a script which gets the list of PRs labeled like that and run `brew pull` on them, so I don't need to copy and paste PR numbers manually.

The API still exposes CI build status so your script could potentially use that. Adding a label as well as status if it makes it easier to script is 👍 to me though.

File: .github/workflows/build_and_upload_bottles.yml (hunk 1 above)
Comment (iMichka) [reply .github/workflows/build_and_upload_bottles.yml]: We could still use the current release pipeline.
Is the bottle downloadable with curl from the github action pipeline?

**Comment by iMichka** [comment]
Another thought on this: can we test building a new llvm bottle with this PR once it is ready. This is needed to test the build performance and upload performance.

**Comment by dawidd6** [comment]
Closing for now.

**Pull request closed without merging**